tests/test_asn1_codec.py

```diff
@@ -0,0 +1,136 @@
+import tuf
+import tuf.util
+import tuf.asn1_codec as asn1_codec
+import tuf.conf
+import unittest
+
+tuf.conf.METADATA_FORMAT = 'json'
+
+class TestASN1Conversion(unittest.TestCase):
+
+
+  # THIS NEXT TEST fails because the TUF root.json test file in question here
+  # uses an RSA key, which the ASN1 conversion does not yet support.
+  # TODO: <~> FIX.
+  # Our ASN1 conversion doesn't seem to support RSA keys. In particular, it is
+  # being assumed that the key values are hex strings ('f9ac1325...') but an
+  # RSA public key value is e.g. '-----BEGIN PUBLIC
+  # KEY------\nMIIBojANBgk...\n...'
+  def test_1_root_partial_convert(self):
```

Comment:
All these test cases fail with a PyAsn1Error exception.  It seems to complain about an incompatible value in the keyid field of the signature.

```Python
======================================================================
ERROR: test_11_root_uptane_partial_convert (__main__.TestASN1Conversion)
Test 11: only_signed conversion PyDict -> ASN1 BER of Root
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_asn1_codec.py", line 66, in test_11_root_uptane_partial_convert
    'tests/repository_data/uptane_mainrepo_root.json', self)
  File "tests/test_asn1_codec.py", line 118, in partial_der_conversion_tester
    role_signable_pydict, only_signed=True)))
  File "/home/vlad/projects/vladforks/tmp/tuf/tuf/asn1_codec.py", line 248, in convert_signed_metadata_to_der
    asn_signed = relevant_asn_module.get_asn_signed(json_signed) # Python3 breaks here.
  File "/home/vlad/projects/vladforks/tmp/tuf/tuf/encoding/rootmetadata.py", line 26, in get_asn_signed
    snapshotPublicKeyid, targetsPublicKeyid, rootMetadata)
  File "/home/vlad/projects/vladforks/tmp/tuf/tuf/encoding/rootmetadata.py", line 123, in set_keys
    tag.tagFormatSimple, 1))
  File "/home/vlad/projects/vladforks/tmp/tuf/env/local/lib/python2.7/site-packages/pyasn1/type/univ.py", line 1945, in __setitem__
    self.setComponentByName(idx, value)
  File "/home/vlad/projects/vladforks/tmp/tuf/env/local/lib/python2.7/site-packages/pyasn1/type/univ.py", line 2037, in setComponentByName
    self._componentType.getPositionByName(name), value, verifyConstraints, matchTags, matchConstraints
  File "/home/vlad/projects/vladforks/tmp/tuf/env/local/lib/python2.7/site-packages/pyasn1/type/univ.py", line 2497, in setComponentByPosition
    raise error.PyAsn1Error('Component value is tag-incompatible: %r vs %r' % (value, componentType))
PyAsn1Error: Component value is tag-incompatible: OctetString(tagSet=TagSet((), Tag(tagClass=128, tagFormat=0, tagId=1)), hexValue='94c836f0c45168f0a437eef0e487b910f58db4d462ae457b5730a4487130f290') vs OctetString(tagSet=TagSet((), Tag(tagClass=128, tagFormat=0, tagId=1)))
```